src/gsi/radarZ_module.f90

```diff
@@ -0,0 +1,2889 @@
+! Calculate the fraction of mleting ice (fr) based on the ratio between
+! qr and qi. fo is the maximum allowable fraction of melting snow.
+!-----------------------------------------------------------------------
+  IF (qr > 0. .AND. qi > 0.) THEN
+    fr = fo*(MIN(qi/qr,qr/qi))**.3
+  ENDIF
+
+!-----------------------------------------------------------------------
+! Calculate the faction of water and ice.
+! fracqr : the mass of water in the melting ice
+! fracqi : the mass of ice in the melting ice
+! fm     : total mass of melting ice
+! fw     : the fraction of water within melting ice
+! rhom   : density of mixture
+!-----------------------------------------------------------------------
+  fracqr = fr * qr
+  fracqi = fr * qi
+  fm = fracqr + fracqi
+
+  IF (fm .EQ. 0. .AND. qr > 0.) THEN
+    fw = 1.
+  ELSE IF (fm > 0.) THEN
+    fw = fracqr/fm
+  ENDIF
+
+  rhom = 1000.*fw**2. + (1.-fw**2.)*density_ice
+
+END SUBROUTINE fractionWater
+
+SUBROUTINE fractionWater_temperature_snow (qr,qi,density_ice,fm,fw,rhom,tair_C)
+
+!-----------------------------------------------------------------------
+!
+! PURPOSE:
+!
+! This subroutine calculates the fractions of water, dry ice (snow or
+! hail), the mixture based on the air temperature.
+! It also calculate the density of mixture.
+!
+!-----------------------------------------------------------------------
+!
+! AUTHOR:  Youngsun Jung, 7/25/2014
+!
+!-----------------------------------------------------------------------
+! Force explicit declarations.
+!-----------------------------------------------------------------------
+
+  IMPLICIT NONE
+
+!-----------------------------------------------------------------------
+! Declare variables.
+!-----------------------------------------------------------------------
+
+  REAL(r_kind) :: qr, qi, density_ice, tair_C
+  REAL(r_kind),INTENT(OUT) :: fm, fw, rhom
+  REAL(r_kind) :: layer_tmax, layer_tmin
+
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+!
+! Beginning of executable code...
+!
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+  fw = 0.
+  fm = 0.
+  rhom = 0.
+
+!-----------------------------------------------------------------------
+! Calculate the faction of water.
+! fm     : total mass of melting ice
+! fw     : the fraction of water within melting ice
+! rhom   : density of mixture
+!-----------------------------------------------------------------------
+
+  fm = qi
+
+! Compute the degree of wet in percentage based on air temperature
+  layer_tmax = 2.5
+  layer_tmin = -2.5
+  if(tair_C >= layer_tmin .and. tair_C < layer_tmax) then
+    fw = (tair_C - layer_tmin)/(layer_tmax-layer_tmin)
+  else if(tair_C >= layer_tmax) then
+    fw = 1.
+  else
+    fm = 0.
+    fw = 0.
+  endif
+
+  rhom = 1000.*fw**2. + (1.-fw**2.)*density_ice
+
+END SUBROUTINE fractionWater_temperature_snow
+
+SUBROUTINE fractionWater_temperature_hail(qr,qi,density_ice,fm,fw,rhom,tair_C)
```

Comment:
The variable, qr, has been identified as unused.  Please remove unused variable.

Reply:
We changed the subroutine to remove qr as you suggested.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-fa0ecb3836c32fcbab1f64a2be08a2292b9c3414d702d475609b32f8205ffcf1R1848